ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactScrollView.java

```diff
@@ -233,55 +239,40 @@ public void getClippingRect(Rect outClippingRect) {
 
   @Override
   public void fling(int velocityY) {
-    if (mScroller != null) {
-      // FB SCROLLVIEW CHANGE
-
-      // We provide our own version of fling that uses a different call to the standard OverScroller
-      // which takes into account the possibility of adding new content while the ScrollView is
-      // animating. Because we give essentially no max Y for the fling, the fling will continue as long
-      // as there is content. See #onOverScrolled() to see the second part of this change which properly
-      // aborts the scroller animation when we get to the bottom of the ScrollView content.
-
-      int scrollWindowHeight = getHeight() - getPaddingBottom() - getPaddingTop();
-
-      mScroller.fling(
-        getScrollX(),
-        getScrollY(),
-        0,
-        velocityY,
-        0,
-        0,
-        0,
-        Integer.MAX_VALUE,
-        0,
-        scrollWindowHeight / 2);
-
-      postInvalidateOnAnimation();
-
-      // END FB SCROLLVIEW CHANGE
+    if (mPagingEnabled) {
+      smoothScrollToPage(velocityY);
     } else {
-      super.fling(velocityY);
-    }
-
-    if (mSendMomentumEvents || isScrollPerfLoggingEnabled()) {
-      mFlinging = true;
-      enableFpsListener();
-      ReactScrollViewHelper.emitScrollMomentumBeginEvent(this);
-      Runnable r = new Runnable() {
-        @Override
-        public void run() {
-          if (mDoneFlinging) {
-            mFlinging = false;
-            disableFpsListener();
-            ReactScrollViewHelper.emitScrollMomentumEndEvent(ReactScrollView.this);
-          } else {
-            mDoneFlinging = true;
-            ReactScrollView.this.postOnAnimationDelayed(this, ReactScrollViewHelper.MOMENTUM_DELAY);
-          }
-        }
-      };
-      postOnAnimationDelayed(r, ReactScrollViewHelper.MOMENTUM_DELAY);
+      if (mScroller != null) {
+        // FB SCROLLVIEW CHANGE
+
+        // We provide our own version of fling that uses a different call to the standard OverScroller
+        // which takes into account the possibility of adding new content while the ScrollView is
+        // animating. Because we give essentially no max Y for the fling, the fling will continue as long
+        // as there is content. See #onOverScrolled() to see the second part of this change which properly
+        // aborts the scroller animation when we get to the bottom of the ScrollView content.
+
+        int scrollWindowHeight = getHeight() - getPaddingBottom() - getPaddingTop();
+
+        mScroller.fling(
+          getScrollX(),
+          getScrollY(),
+          0,
+          velocityY,
+          0,
+          0,
+          0,
+          Integer.MAX_VALUE,
+          0,
+          scrollWindowHeight / 2);
+
+        postInvalidateOnAnimation();
+
+        // END FB SCROLLVIEW CHANGE
+      } else {
+        super.fling(velocityY);
```

Comment:
Is this the code moved from above? Looks like you also refactored it but from the GitHub UI I can't tell what changes you made exactly. Would be helpful to add some inline comments on this PR.